timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/TimeLockServerConfiguration.java

```diff
@@ -76,4 +84,27 @@ public ClusterConfiguration cluster() {
     public long slowLockLogTriggerMillis() {
         return 10000;
     }
+
+    public boolean useThreadPooling() {
+        return useThreadPooling;
+    }
+
+    public int availableThreads() {
+        if (!useThreadPooling()) {
+            throw new IllegalStateException("Should not call availableThreads() if useThreadPooling is disabled");
+        }
+
+        return computeNumberOfAvailableThreads();
+    }
+
+    private int computeNumberOfAvailableThreads() {
+        DefaultServerFactory serverFactory = (DefaultServerFactory) getServerFactory();
+        int maxServerThreads = serverFactory.getMaxThreads();
+        HttpConnectorFactory connectorFactory = (HttpConnectorFactory) serverFactory.getApplicationConnectors().get(0);
```

Comment:
The answer to this question is maybe:

- the TimeLockServerConfiguration has a `setServerFactory` method. If this method is used to set other instances of ServerFactory which are not a DefaultServerFactory, we would have a ClassCastException on line 101;
- the DefaultServerFactory also has a `setApplicationConnectors`. If set to an empty list, or null object, line 103 will trigger exceptions.

The default behavior won't cause exceptions. Ideally, we would have other ways of accessing maxServerThreads, selectorThreads and acceptorThreads instead of trusting the default behavior. We can discuss better runtime checks for this later, but I see no way of adding compile time checks. Also, maybe we want to follow up with another PR to add those checks on the setters themselves.